src/devices/Button/GpioButton.cs

```diff
@@ -13,55 +13,81 @@ namespace Iot.Device.Button
-        public GpioButton(int buttonPin, GpioController? gpio = null, bool shouldDispose = true, PinMode pinMode = PinMode.InputPullUp,
-            TimeSpan debounceTime = default(TimeSpan))
-            : this(buttonPin, TimeSpan.FromTicks(DefaultDoublePressTicks), TimeSpan.FromMilliseconds(DefaultHoldingMilliseconds), gpio, shouldDispose, pinMode, debounceTime)
+        public GpioButton(int buttonPin, bool isPullUp = true, bool hasExternalResistor = false,
+            GpioController? gpio = null, bool shouldDispose = true, TimeSpan debounceTime = default)
+            : this(buttonPin, TimeSpan.FromTicks(DefaultDoublePressTicks), TimeSpan.FromMilliseconds(DefaultHoldingMilliseconds), isPullUp, hasExternalResistor, gpio, shouldDispose, debounceTime)
         {
         }
 
         /// <summary>
         /// Initialization of the button.
         /// </summary>
         /// <param name="buttonPin">GPIO pin of the button.</param>
-        /// <param name="pinMode">Pin mode of the system.</param>
         /// <param name="doublePress">Max ticks between button presses to count as doublepress.</param>
         /// <param name="holding">Min ms a button is pressed to count as holding.</param>
+        /// <param name="isPullUp">True if the Gpio is either pulled up in hardware or in the Gpio configuration (see <paramref name="hasExternalResistor"/>. False if instead the Gpio is pulled down.</param>
+        /// <param name="hasExternalResistor">When False the pull resistor is configured using the Gpio PinMode.InputPullUp or PinMode.InputPullDown (if supported by the board). Otherwise the Gpio is configured as PinMode.Input.</param>
         /// <param name="gpio">Gpio Controller.</param>
         /// <param name="shouldDispose">True to dispose the GpioController.</param>
         /// <param name="debounceTime">The amount of time during which the transitions are ignored, or zero</param>
-        public GpioButton(int buttonPin, TimeSpan doublePress, TimeSpan holding, GpioController? gpio = null, bool shouldDispose = true, PinMode pinMode = PinMode.InputPullUp, TimeSpan debounceTime = default(TimeSpan))
+        public GpioButton(int buttonPin,
+            TimeSpan doublePress,
+            TimeSpan holding,
+            bool isPullUp = true,
+            bool hasExternalResistor = false,
+            GpioController? gpio = null,
+            bool shouldDispose = true,
+            TimeSpan debounceTime = default)
             : base(doublePress, holding, debounceTime)
         {
             _gpioController = gpio ?? new GpioController();
-            _shouldDispose = shouldDispose;
+            _shouldDispose = gpio == null ? true : shouldDispose;
             _buttonPin = buttonPin;
-            _pinMode = pinMode;
-            _debounceTime = debounceTime;
+            HasExternalResistor = hasExternalResistor;
+
+            _eventPinMode = isPullUp ? PinMode.InputPullUp : PinMode.InputPullDown;
```

Comment:
This is dangerous! Because some board do not support PullUp/PullDown. It's the reason why originally you have to pass the pull type.
If you want to do something like this, you need first to check if the pin mode is supported, if not, then choose one that is a default one.
See here for the good way to do it: https://github.com/dotnet/iot/blob/main/src/devices/Dhtxx/DhtBase.cs#L192

Reply:
@Ellerbach I am aware that not every pin can be internally pulled up or down.
I didn't change the logic of the original author, I just added support for the external pullup/down resistor.

I was not aware of `IsPinModeSupported` in `GpioController`. Good to know, Now I can throw an exception if it's not.

Reply:
I revised two cases:
- if the button can be configured as input (certain gpio can only be output)
- if the button can be configured with the configuration deriving from `isPullUp` and `HasExternalResistor`

Reply:
I think this is fine. If the mode is not supported, it will throw, which is expected, because it's an user error.